server/src/com/cloud/network/router/CommandSetupHelper.java

```diff
@@ -778,10 +778,11 @@ public int compare(final PublicIpAddress o1, final PublicIpAddress o2) {
 
                 final boolean add = ipAddr.getState() == IpAddress.State.Releasing ? false : true;
                 boolean sourceNat = ipAddr.isSourceNat();
-                /* enable sourceNAT for the first ip of the public interface */
-                if (firstIP) {
-                    sourceNat = true;
+                /* enable sourceNAT for the first ip of the public interface as long as it's source nat. */
+                if (firstIP && !sourceNat) {
```

Comment:
How do I trigger this case using the CS API? Is it by adding multiple NICs to the virtual router?